csharp/extractor/Semmle.Extraction.CSharp/Entities/Types/Type.cs

```diff
@@ -160,6 +168,22 @@ protected void PopulateType(TextWriter trapFile, bool constructUnderlyingTupleTy
                     trapFile.parent_namespace(this, Namespace.Create(Context, ns));
             }
 
+            if (Symbol.IsInlineArray())
+            {
+                // Roslyn enforces that "inline arrays" (decorated structs)
+                // only contain a single (instance) field, which is used to capture
+                // the type of the elements.
+                var elementType = Symbol
+                    .GetMembers()
+                    .OfType<IFieldSymbol>()
+                    .Where(field => !field.IsStatic && !field.IsConst)
+                    .Single()
+                    .Type;
+
+                var t = Create(Context, elementType);
+                trapFile.inline_array_element_type(this, t);
```

Comment:
Couldn't we also extract the length of the inline array?

Reply:
Yes, that would make sense!